src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -840,7 +840,7 @@ class InsertStatementSegment(BaseSegment):
 
 @spark3_dialect.segment()
 class InsertOverwriteDirectorySegment(BaseSegment):
-    """An `INSERT OVERWRITE [LOCAL] DIRECTORY` statement to insert or overwrite new rows into a table.
+    """An `INSERT OVERWRITE [LOCAL] DIRECTORY` statement to insert or overwrite new rows into a directory path.
```

Comment:
Seems a lot of repetition here (and also doing a lot of work on trying to limit our line count in #2264).

```suggestion
    """An `INSERT OVERWRITE [LOCAL] DIRECTORY` statement.
```